csrc/fusion_segmenter.cpp

```diff
@@ -1868,13 +1868,20 @@ std::pair<IrCloner, std::unique_ptr<Fusion>> SegmentedFusion::makeFusion(
     fusion_segment->removeOutput(out);
   }
 
-  std::vector<TensorView*> view_tvs;
+  std::vector<TensorView*> sf_tvs;
   for (auto inp : getAllInputs(sg)) {
     auto clone_tv = complete_to_segment_map.clone(inp);
     fusion_segment->addInput(clone_tv);
     if (inp->isDefinitionType<ReshapeOp>()) {
       NVF_ERROR(clone_tv != nullptr && clone_tv->isA<TensorView>());
-      view_tvs.push_back(clone_tv->as<TensorView>());
+    } else if (inp->isDefinitionType<PreprocessGroupedMatmulInputSf>()) {
+      // There's no point of replaying allocation domain if we cannot index into
```

Comment:
I see your point, but this seems rather fragile, and I'm lost how all the allocation domain tricks are supposed to work together. The proper allocation domain is still necessary at the run time when allocating the output, right? But the fusion no longer has it. How would everything be piped together?

Reply:
> The proper allocation domain is still necessary at the run time when allocating the output, right? But the fusion no longer has it. How would everything be piped together?

The trick here is that, we are only doing this for fusion inputs (consumer segment of the output of `PreprocessGroupedMatmulInputSf`)

In the segment that contains the `PreprocessGroupedMatmulInputSf` op, we are not changing anything, so it would still keep the padded allocation domain, so output would be allocated with the right size.

For consumer of this operation, we would erase its allocation domain, because we are not indexing, nor are we allocating it.

^^^ Does this make sense to you? I'll add them to the code comment.